samtranslator/schema/schema.py

```diff
@@ -51,6 +51,65 @@ class AwsServerlessConnector(BaseModel):
     Properties: ConnectorProperties
 
 
+class EventsDeadLetterConfig(BaseModel):
+    Arn: Optional[PassThrough]
+    QueueLogicalId: Optional[str]
+    Type: Optional[Literal["SQS"]]
+
+
+class EventsScheduleProperties(BaseModel):
+    DeadLetterConfig: Optional[EventsDeadLetterConfig]
+    Description: Optional[PassThrough]
+    Enabled: Optional[bool]
+    Input: Optional[PassThrough]
+    Name: Optional[PassThrough]
+    RetryPolicy: Optional[PassThrough]
+    Schedule: Optional[PassThrough]
+    State: Optional[PassThrough]
+
+
+class EventsSchedule(BaseModel):
+    Type: Literal["Schedule"]
+    Properties: EventsScheduleProperties
+
+
+class EventsScheduleV2Properties(BaseModel):
+    DeadLetterConfig: Optional[EventsDeadLetterConfig]
+    Description: Optional[PassThrough]
+    EndDate: Optional[PassThrough]
+    FlexibleTimeWindow: Optional[PassThrough]
+    GroupName: Optional[PassThrough]
+    Input: Optional[PassThrough]
+    KmsKeyArn: Optional[PassThrough]
+    Name: Optional[PassThrough]
+    PermissionsBoundary: Optional[PassThrough]
+    RetryPolicy: Optional[PassThrough]
+    RoleArn: Optional[PassThrough]  # TODO: Add to docs
+    ScheduleExpression: Optional[PassThrough]
+    ScheduleExpressionTimezone: Optional[PassThrough]
+    StartDate: Optional[PassThrough]
+    State: Optional[PassThrough]
+
+
+class EventsScheduleV2(BaseModel):
+    Type: Literal["ScheduleV2"]
+    Properties: EventsScheduleV2Properties
+
+
+class ResourcePolicy(BaseModel):
```

Comment:
nit: there's always a little risk of sharing models unless we're sure they share it by design (and do not just currently _happen_ to have the same properties), as new properties will be added to both and tests will pass even if it was meant for only one of them